pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -452,6 +453,67 @@ func (l *LibvirtDomainManager) asyncMigrate(vmi *v1.VirtualMachineInstance, opti
 	}(l, vmi)
 }
 
+func (l *LibvirtDomainManager) SetGuestTime(vmi *v1.VirtualMachineInstance) error {
+	// Try to set VM time to the current value.  This is typically useful
+	// when clock wasn't running on the VM for some time (e.g. during
+	// suspension or migration), especially if the time delay exceeds NTP
+	// tolerance.
+	// It is not guaranteed that the time is actually set (it depends on guest
+	// environment, especially QEMU agent presence) or that the set time is
+	// very precise (NTP in the guest should take care of it if needed).
+
+	domName := api.VMINamespaceKeyFunc(vmi)
+	dom, err := l.virConn.LookupDomainByName(domName)
+	if err != nil {
+		log.Log.Object(vmi).Reason(err).Error("failed to sync guest time")
+		return err
+	}
+	// try setting the guest time for 5 seconds. Fail early if agent is not configured.
+	go l.setGuestTime(vmi, dom)
+	return nil
+}
+
+func (l *LibvirtDomainManager) setGuestTime(vmi *v1.VirtualMachineInstance, dom cli.VirDomain) {
+	timeout := time.After(60 * time.Second)
+	tick := time.Tick(1 * time.Second)
```

Comment:
I think that you have to stop the ticker, right?

Reply:
Yep. Done